### PR TITLE
Non Destructive ERM for User Roles

### PR DESCRIPTION
**Pull request by mjriley**
## Product Description
This is the first part of non-destructive push for ERM. This initial PR is just for handling User Roles. Note that this is currently incomplete, as we're awaiting decisions on both the language to best notify users of how their data will be synced (currently, it claims everything will be overwritten, which isn't correct), as well as how to handle 'orphaned' roles, where the upstream domain has removed a role that was synced to a downstream domain.

## Technical Summary
The ERM work is tracked [here](https://dimagi-dev.atlassian.net/browse/SAAS-13621)

### Safety story
These changes have been tested locally and through automated suites. I will be putting this on staging for additional testing/acceptance.

### Automated test coverage

The new python functionality is covered by unit or integration tests. The javascript and HTML changes are uncovered.

### QA Plan

No QA needed.

### Rollback instructions

- [x] This PR can be reverted after deploy with no further considerations

### Labels & Review
- [x] Risk label is set correctly
- [x] The set of people pinged as reviewers is appropriate for the level of risk of the change

### CODE REVIEW

File: corehq/apps/users/tests/test_views.py [hunk 1]
```diff
@@ -174,6 +176,78 @@ def test_cannot_update_parent_data(self):
...
+
+    def test_cannot_delete_linked_domain_data(self):
+        role = UserRole.objects.create(domain='test-domain', name='test-role', upstream_id='parent_id')
+
+        request = self._create_request({'_id': role.get_id})
+
+        response = delete_user_role(request, 'test-domain')
+        remaining_role = UserRole.objects.get(id=role.id)
+        # Following existing behavior. This should return an error when the missing role also returns an error
+        self.assertEqual(response.content.decode('utf-8'), '{}')
```
Comment (gherceg) [inline corehq/apps/users/tests/test_views.py:205]: I know you are following existing behavior, but seeing as this part is new code you added in this PR, why not address the response as part of this PR?

**Review by gherceg — COMMENTED** [review]
Looks good so far. Maybe you are waiting on language to be updated, but it would be nice to see a screenshot of the updated UI once that is complete.

File: corehq/apps/users/tests/test_views.py (hunk 1 above)
Comment (mjriley) [reply corehq/apps/users/tests/test_views.py:205]: Most likely will be doing so

File: corehq/apps/users/templates/users/roles_and_permissions.html [hunk 2]
```diff
@@ -118,6 +118,9 @@ <h3 class="panel-title">
           </th>
           {% endif %}
           <th>{% trans "Actions" %}</th>
+          {% if is_managed_by_upstream_domain %}
+          <th>{% trans "Local App Config" %}</th>
```
Comment (esoergel) [inline corehq/apps/users/templates/users/roles_and_permissions.html:122]: Maybe "Local Project Config" or "Locally configured"?  This identifies which roles are local and can thus be managed, right?

**Review by esoergel — APPROVED** [review]

**Comment by mjriley** [comment]
Closing, as did not recall having put this up for PR. This will be re-opened once the full ERM changes are ready for review

**Pull request closed without merging**